awscli/customizations/s3/s3handler.py

```diff
@@ -471,6 +472,13 @@ def _submit_transfer_request(self, fileinfo, extra_args, subscribers):
         bucket, key = find_bucket_key(fileinfo.dest)
         source_bucket, source_key = find_bucket_key(fileinfo.src)
         copy_source = {'Bucket': source_bucket, 'Key': source_key}
+        if not self._should_transfer_begin(fileinfo):
```

Comment:
Could this get registered in `_get_warning_handlers` instead of modifying this method?